addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1308,223 +1308,266 @@ registry.background = SnippetOption.extend({
 });
 
 /**
- * Handles the edition of snippet's background image position.
+ * Handles the edition of snippets' background image position.
  */
-registry['background_position'] = SnippetOption.extend({
+registry.backgroundPosition = SnippetOption.extend({
     xmlDependencies: ['/web_editor/static/src/xml/editor.xml'],
 
     /**
      * @override
      */
     start: function () {
         this._super.apply(this, arguments);
-        var self = this;
-        this.$target.on('snippet-option-change', function () {
-            self.onFocus();
-        });
+        this.$target.on('snippet-option-change', () => this._updateUI());
+
+        this._initOverlay();
+        const self = this;
+        this.backgroundSize = {
+            get x() {
+                return self.$target.attr('data-bg-size-x') || 100;
+            },
+            set x(value) {
+              self.$target.attr('data-bg-size-x', value);
+              self.$target.css({backgroundSize: `${this.x}% ${this.y}%`});
+            },
+            get y() {
+                return self.$target.attr('data-bg-size-y') || 100;
+            },
+            set y(value) {
+              self.$target.attr('data-bg-size-y', value);
+              self.$target.css({backgroundSize: `${this.x}% ${this.y}%`});
+            },
+        };
+        // Fill inputs with current values.
+        this.$el.find('.o_bg_width_input input').val(this.backgroundSize.x);
+        this.$el.find('.o_bg_height_input input').val(this.backgroundSize.y);
```

Comment:
Let's do what we talked about:
- The background width in pixels (nothing being auto)
- Forget about saving the values when switching to cover... or do it internally in the option for the time of the edition (not writing the DOM)
- Enable your code so that the drag for the position is more coherent

Reply:
Alright, it's done. Background size in pixels, using the `we-input`'s `data-set-style`, no longer saving size values, and dragging is synchronized between the mouse and background. Also rebased on top of your latest changes and everything seems to be working fine.